packages/common/base/src/apiClient/ApiClient.ts

```diff
@@ -0,0 +1,41 @@
+export abstract class ApiClient {
+    #authHeaders: HeadersInit;
+    constructor(authHeaders: HeadersInit) {
+        this.#authHeaders = authHeaders;
+    }
+
+    abstract get baseUrl(): string;
+
+    private async makeRequest(path: string, init: RequestInit) {
+        return fetch(`${ApiClient.normalizePath(this.baseUrl)}/${ApiClient.normalizePath(path)}`, {
+            ...init,
+            headers: { ...this.#authHeaders, ...init.headers }, // #authHeaders intentionally first, in case sub class wants to override
+        });
+    }
+
+    async get(path: string, params: Omit<RequestInit, "method">) {
+        return this.makeRequest(path, { ...params, method: "GET" });
+    }
+
+    async post(path: string, params: Omit<RequestInit, "method">) {
+        return this.makeRequest(path, { ...params, method: "POST" });
+    }
+
+    async put(path: string, params: Omit<RequestInit, "method">) {
+        return this.makeRequest(path, { ...params, method: "PUT" });
+    }
+
+    async delete(path: string, params: Omit<RequestInit, "method">) {
+        return this.makeRequest(path, { ...params, method: "DELETE" });
+    }
+
+    async patch(path: string, params: Omit<RequestInit, "method">) {
+        return this.makeRequest(path, { ...params, method: "PATCH" });
+    }
```

Comment:
eventually we probably want these functions to be more like the ones in crossmint-main, for ex with retries + json parsng, but this is just the strart